lib/AST/ASTMangler.cpp

```diff
@@ -1233,7 +1233,12 @@ void ASTMangler::appendType(Type type, GenericSignature sig,
 
     case TypeKind::ExistentialMetatype: {
       ExistentialMetatypeType *EMT = cast<ExistentialMetatypeType>(tybase);
-      appendType(EMT->getInstanceType(), sig, forDecl);
+      if (EMT->getInstanceType()->isExistentialType() &&
+          EMT->hasParameterizedExistential())
+        appendConstrainedExistential(EMT->getInstanceType(), sig, forDecl);
```

Comment:
Metatypes can be nested

Reply:
`EMT->getInstanceType()->isExistentialType()` ensures that we don't descend into them. `ExistentialMetatype` is only considered by `isAnyExistentialType()`.

Reply:
Okay, so if we don't need to descend into nested metatypes here, why do we need to call out direct existential metatypes at all instead of just letting the recursive visitation in `appendType` handle it?

Reply:
Like, my patch to symbolically reference shapes does need to do something at this level because the existential metatype is part of the shape, but I think the normal mangling path doesn't have to.  You'll just get down to the `ExistentialType` and do what you need to do at that level.

Reply:
> why do we need to call out direct existential metatypes at all instead of just letting the recursive visitation in appendType handle it?

The AST for `any P<T>.Type` currently looks like `(ExistentialMetatypeType (ParameterizedProtocolType ...))` which is unfortunate...